toolchain/parse/testdata/operators/recover_star_minus.carbon

```diff
@@ -24,7 +24,7 @@ var n: i8 = n* -n;
 // CHECK:STDOUT:           {kind: 'IdentifierNameExpr', text: 'n'},
 // CHECK:STDOUT:         {kind: 'PostfixOperatorStar', text: '*', subtree_size: 2},
 // CHECK:STDOUT:         {kind: 'IdentifierNameExpr', text: 'n'},
-// CHECK:STDOUT:       {kind: 'InfixOperatorMinus', text: '-', has_error: yes, subtree_size: 4},
+// CHECK:STDOUT:       {kind: 'InfixOperatorMinus', text: '-', subtree_size: 4},
```

Comment:
This definitely seems like a regression.

Reply:
Before, there was no precedence defined between a binary `-` and unary `T*`. This can be seen in:
https://github.com/carbon-language/carbon-lang/blob/4132c6a65f891c9eabb937abfcfc2adc5cf45936/docs/design/expressions/README.md

There is now a precedence defined between them. Specifically, "T*" is higher than "x * y and x / y" is higher than "x + y and x - y". This can be seen in:
https://github.com/carbon-language/carbon-lang/blob/trunk/docs/design/expressions/README.md

Since there is now a defined precedence between these operators, the parse is well-defined.

Reply:
The leads accepted this regression as part of the changes they wanted in #4075 . I don't think I captured that explicitly in the proposal, but I recall it from the discussion.